packages/zen-bull-netting/src/NettingLib.sol

```diff
@@ -6,186 +6,296 @@ import { IERC20 } from "openzeppelin/interfaces/IERC20.sol";
+     * @param _oSqth oSQTH address
+     * @param _trader market maker address
+     * @param _bidId MM's bid ID
+     * @param _oSqthBalance remaining netting contracts's oSQTH balance
+     * @param _quantity oSQTH quantity in market maker order
+     */
+    function transferOsqthToMarketMakers(
+        address _oSqth,
+        address _trader,
+        uint256 _bidId,
+        uint256 _oSqthBalance,
+        uint256 _quantity
+    ) external returns (bool, uint256) {
+        uint256 remainingOsqthBalance;
+        if (_quantity < _oSqthBalance) {
+            IERC20(_oSqth).transfer(_trader, _quantity);
+
+            remainingOsqthBalance = _oSqthBalance - _quantity;
+
+            emit TransferOsqthToMarketMakers(_trader, _bidId, _quantity, remainingOsqthBalance);
+
+            return (false, remainingOsqthBalance);
         } else {
-            IERC20(oSqth).transferFrom(trader, address(this), oSqthRemaining);
-            return (true, 0);
+            IERC20(_oSqth).transfer(_trader, _oSqthBalance);
+
+            emit TransferOsqthToMarketMakers(_trader, _bidId, _oSqthBalance, remainingOsqthBalance);
+
+            return (true, remainingOsqthBalance);
         }
     }
 
-    function transferOsqthToMarketMakers(
-        address oSqth,
-        address trader,
-        uint256 bidId,
-        uint256 oSqthBalance,
-        uint256 quantity
-    ) external returns (bool, uint256) {
-        if (quantity < oSqthBalance) {
-            IERC20(oSqth).transfer(trader, quantity);
+    /**
+     * @notice transfer oSQTH from market maker
+     * @dev this is executed during the withdraw auction, MM selling OSQTH for WETH
+     * @param _oSqth oSQTH address
+     * @param _trader market maker address
+     * @param _totalOsqthToPull total amount of oSQTH to transfer from order array
+     * @param _quantity oSQTH quantity in market maker order
+     */
+    function transferOsqthFromMarketMakers(
+        address _oSqth,
+        address _trader,
+        uint256 _totalOsqthToPull,
+        uint256 _quantity
+    ) internal returns (bool, uint256) {
+        uint256 oSqthRemaining;
+        if (_quantity < _totalOsqthToPull) {
+            IERC20(_oSqth).transferFrom(_trader, address(this), _quantity);
+
+            oSqthRemaining = _totalOsqthToPull - _quantity;
 
-            emit DepositAuction(trader, bidId, quantity);
+            emit TransferOsqthFromMarketMakers(_trader, _quantity, oSqthRemaining);
 
-            return (false, (oSqthBalance - quantity));
+            return (false, oSqthRemaining);
         } else {
-            IERC20(oSqth).transfer(trader, oSqthBalance);
+            IERC20(_oSqth).transferFrom(_trader, address(this), _totalOsqthToPull);
 
-            emit DepositAuction(trader, bidId, oSqthBalance);
+            emit TransferOsqthFromMarketMakers(_trader, _totalOsqthToPull, oSqthRemaining);
 
-            return (true, 0);
+            return (true, oSqthRemaining);
```

Comment:
it doesn't really matter, but this should return (true, 0) for consistency imo

Reply:
I actually don't even think we need these break function or to return true/false because the for loop has break if toExchange = 0, more notes in the general review comments.